hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/lock/OzoneManagerLock.java

```diff
@@ -486,11 +502,12 @@ private void updateReadUnlockMetrics(Resource resource, String resourceName) {
       // Adds a snapshot to the metric readLockHeldTimeMsStat.
       omLockMetrics.setReadLockHeldTimeMsStat(
           TimeUnit.NANOSECONDS.toMillis(readLockHeldTimeNanos));
+      updateProcessingDetails(Timing.LOCKSHARED, readLockHeldTimeNanos);
```

Comment:
I feel like this updateReadUnlockMetrics and updateWriteUnlockMetrics lock held time is inaccurate, for the hold time is saved in Resource object which is shared between different resoucesNames.  Say the Resource is VOLUME_LOCK, if two clients, each access a different volume concurrently, then this lock hold time will be calculated by (latest lock release time - earliest lock lock time).  
@tanvipenumudy , could you help to check it if the above is true or not?

Reply:
Given currently the OMStateMachine has single thread to execute all write requests, this inaccurate will not happen. Future, if the single thread is optimized to multi threads, then this part need a revisit.